icees_api/features/sql.py

```diff
@@ -946,6 +949,31 @@ def validate_range(conn, table_name, feature):
         print(f"warning: cannot validate feature {feature_name} in table {table_name} because its levels are not provided")
 
 
+def validate_feature_value_in_table_column_for_equal_operator(conn, table_name, feature):
```

Comment:
@maximusunc I found number 1.0 is treated as string when doing select with where equal comparison in sqlite. Specifically, number 1.0 input returns the same output as string 1.0, so validating type to make sure it is string does not make much sense here since sqlite query returns the same result regardless whether the input is number 1.0 or string 1.0. So after giving this some thought, I think for equal operator, it is good to validate the value is in the database since otherwise, the opposite `<>` operator to the equal operator just catch every record in the database if the value does not exist in the database in the first place. I believe this is the main point in the issue #217 since the response result does not make sense. My fix in this PR will address this and inform the user the wrong input value in the query. This validation only apply to the equal operator since for other operators it does not make sense to make sure the value is in the database. I checked various scenario and found adding this limited validation worked pretty well to address the concerns detailed in the issue #217. Hopefully this explanation makes sense to you, but let me know if not.

Reply:
Why not just check if the string input is a number/float? Then you don't have to query the database at all. A performant solution is https://stackoverflow.com/a/23639915

Reply:
@maximusunc Great point! I have added the check for number/float which simplified two potential queries into one. I did run into some problem with the change since a non-number string such as Female in the database will be casted as 0, but I have resolved that problem. See if it looks good now. As a clarification just in case, I do have to query the database for validation in order to check whether the input query value with equal operation exists in the database.